### ISSUE TITLE
AMD FSR(FidelityFX Super Resolution) Feature Request

### ISSUE THREAD

**Issue by xlen** [hrydgard/ppsspp#14689]
### What should happen

I'd really love to see an implementation of AMD's FSR(FidelityFX Super Resolution).

### Who would this benefit

Everyone could benefit from it, better performance/wider performance scaling options on any device.

### Games this would be useful in

Pretty much anything

### Other emulators or software with a similar feature

Xenia and RPCS3 have already WIP builds with FSR, Wine, Proton, and about 40 windows games have FSR support as well.

### Checklist

- [X] Check the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by hrydgard** [comment]
Since PSP rendering is a lot lighter on the GPU than either Xenia or RPCS3, I expect the gains from upscaling a smaller resolution here to be much smaller than it is in those emulators, if any. However, might still be worth a try at some point and see how it looks.

**Comment by Gamemulatorer** [comment]
Related https://github.com/hrydgard/ppsspp/issues/14630

**Comment by ANR2ME** [comment]
Hopefully, FSR can upscale the framebuffer (and textures) a bit better in quality may be :)

**Comment by Leopard20** [comment]
> 
> 
> Since PSP rendering is a lot lighter on the GPU than either Xenia or RPCS3, I expect the gains from upscaling a smaller resolution here to be much smaller than it is in those emulators, if any. However, might still be worth a try at some point and see how it looks.

I wonder if FSR does anything to 2D games?

**Comment by xlen** [comment]
> 
> 
> > Since PSP rendering is a lot lighter on the GPU than either Xenia or RPCS3, I expect the gains from upscaling a smaller resolution here to be much smaller than it is in those emulators, if any. However, might still be worth a try at some point and see how it looks.
> 
> I wonder if FSR does anything to 2D games? (better quality than current postprocessing methods?)

yeah, FSR doesn't really care about dimensions, it's essentially upscaling the whole frame while taking edges into consideration. it might be a better option than the conventional upscaling methods in most cases I've seen up to date.

**Comment by xlen** [comment]
> 
> 
> Since PSP rendering is a lot lighter on the GPU than either Xenia or RPCS3, I expect the gains from upscaling a smaller resolution here to be much smaller than it is in those emulators, if any. However, might still be worth a try at some point and see how it looks.

I was thinking about that, but there are 2 things to note, most people playing are upscaling in multiples of what the native PSP resolution was and FSR can be useful to limit the resources needed especially on mobile while also cleaning up aliasing artifacts, alternatively CAS/RCAS could be an interesting sharpening option for cleaning up the image as well

**Comment by xlen** [comment]
> 
> 
> Related #14630

Interestingly, the search didn't find anything directly related to FSR, might be a GitHub-related issue? 👀

**Comment by ss8688** [comment]
Using https://github.com/Blinue/Magpie can make fsr work with ppsspp
Yes,although ppsspp do not support fsr

**Comment by ss8688** [comment]
It seems that fsr only can upscale 3D graphic？

**Comment by xlen** [comment]
> 
> 
> It seems that fsr only can upscale 3D graphic？

nope, FSR doesn't really care, you can use it for pretty much anything

**Comment by justaguyonthehub** [comment]
With FSR 3's frame generation, you could make games that are hard-locked to 30 or 60 FPS look like 60 or 120 FPS. Not every game has a frame rate unlocker patch (due to physics and/or animations, i think). FSR 3.1 would also be an option for those who want to upscale games with low-res assets, in addition to the texture filter options. I hope it can be revisited.

**Comment by LunaMoo** [comment]
If you want 30 fps games to appear as 120 fps right now you can just just use a software like "loseless scaling", through with fake frames in games, the results are pretty poor when you start at 30 or lower fps. At such framerates fake frames only look decent in videos as it's already tech built around key frames and lying to our brains, with games you get a lot of noticeable artifacts and smoothing it's said games need at minimum 45, but based on my own experience with pc games I'd up that even more and say you need to start with 60 fps for fake frames to show decent results, so you'd still need game patches.

 FSR wouldn't really work for upscaling assets in PSP games either simply because it's games assets are just too low res to begin with.

 It'll also not improve performance in PSP games on weak devices as Henrik explained.


 Overall PSP games doesn't provide enough information for those faking algorithms to give a decent results and wouldn't really provide a solution for low end hardware owners either as such this feature isn't really that interesting as it appears following the trends in PC gaming, but PPSSPP is open source so you're welcome to implement it on your own if you value it enough to spend your own time on implementing it.

**Comment by justaguyonthehub** [comment]
I've already been using AMD Fluid Motion Frames to make Hot Shots Tennis look 60fps, and the results have been pretty good at 1440p. But I can't speak for high-paced action games.

Recently, I realized that when I move to Linux, I wont be able to use Fluid Motion Frames because it's a Windows exclusive feature of the AMD software. So the only way to use frame gen in games/emulators were if it was implemented natively. Just another reason I think this feature should be implemented.

Unfortunately, I'm not a dev. I'm just making the suggestion. If I could have it my way, every game would have a patch that unlocks FPS fully without any problems, but that seems way harder to implement even on a game-by-game basis

**Comment by hrydgard** [comment]
Well, it would be cool to try it, it's just quite a bit of work to integrate it, and I can't prioritize it right now.
